std/algorithm/searching.d

```diff
@@ -4192,9 +4192,24 @@ bool startsWith(alias pred = "a == b", R, E)(R doesThisStart, E withThis)
 if (isInputRange!R &&
     is(typeof(binaryFun!pred(doesThisStart.front, withThis)) : bool))
 {
-    return doesThisStart.empty
-        ? false
-        : binaryFun!pred(doesThisStart.front, withThis);
+    if (doesThisStart.empty)
+        return false;
+
+    alias predFunc = binaryFun!pred;
+
+    // auto-decoding special case
+    static if (isNarrowString!R)
+    {
+        // specialize for ASCII as to not change previous behavior
+        if (withThis <= 0x7F)
+            return predFunc(doesThisStart[0], withThis);
```

Comment:
Sorry, I missed that.

> Can't, that would change the behavior of the function when dealing with multi code unit code points.

Can you explain how? If you're searching for something potentially smaller than a code-point, why would you want autodecoding? Furthermore, if the `withThis` is a surrogate pair, you will never find it.